File: typescript/design-patterns/facade/facade/HomeTheaterFacade.ts
// One simple method that orchestrates the mess.
class HomeTheaterFacade {
  constructor(
    private amp = new Amplifier(),
    private lights = new Lights(),
    private projector = new Projector(),
    private box = new StreamingBox(),
  ) {}

  watchMovie(title: string) {
    this.lights.dim(10);
    this.projector.on();
    this.projector.wide();
    this.amp.on();
    this.amp.setVolume(7);
    this.box.play(title);          // 6 calls → one for the caller
  }
}

File: java/design-patterns/facade/facade/HomeTheaterFacade.java
class HomeTheaterFacade {
    private final Amplifier amp = new Amplifier();
    private final Lights lights = new Lights();
    private final Projector projector = new Projector();
    private final StreamingBox box = new StreamingBox();

    public void watchMovie(String title) {
        lights.dim(10);
        projector.on();
        projector.wide();
        amp.on();
        amp.setVolume(7);
        box.play(title);
    }
}